src/node/nodestate.h

```diff
@@ -61,8 +102,11 @@ namespace ccf
     StateMachine(T s) : s(s) {}
     void expect(T s) const
     {
-      if (s != this->s.load())
-        throw std::logic_error("Unexpected state");
+      auto state = this->s.load();
+      if (s != state)
+      {
+        throw std::logic_error(fmt::format("Unexpected state: {}", state));
```

Comment:
Message should include the _expected_ state as well as the current state.

Reply:
That's much easier to infer from the trace, but sure, why not!